torch/csrc/jit/codegen/cuda/type.h

```diff
@@ -54,12 +54,25 @@ enum class PredicateType {
   ReductionWrite
 };
 
-enum class DataType { Double, Float, Half, Int, Int32, Bool, BFloat16, Null };
+enum class DataType {
+  Double,
+  Float,
+  Half,
+  Int,
+  Int32,
+  Bool,
+  BFloat16,
+  ComplexFloat,
```

Comment:
Just curious, is complex float and complex double the only complex types we will have?

Reply:
There is also a `ComplexHalf`, but PyTorch has not started implementing any op for it, even in eager mode. So for now I just consider this as non-existent.